cmd/ndm_daemonset/probe/eventhandler.go

```diff
@@ -39,29 +46,49 @@ type ProbeEvent struct {
 
 // addBlockDeviceEvent fill block device details from different probes and push it to etcd
 func (pe *ProbeEvent) addBlockDeviceEvent(msg controller.EventMessage) {
-	bdList, err := pe.Controller.ListBlockDeviceResource()
+	bdAPIList, err := pe.Controller.ListBlockDeviceResource(true)
 	if err != nil {
 		klog.Error(err)
 		go pe.initOrErrorEvent()
 		return
 	}
+
+	isGPTBasedUUIDEnabled := pe.Controller.FeatureGates.IsEnabled(features.GPTBasedUUID)
+
 	isErrorDuringUpdate := false
 	// iterate through each block device and perform the add/update operation
 	for _, device := range msg.Devices {
-		klog.Infof("Processing details for %s", device.UUID)
+		klog.Infof("Processing details for %s", device.DevPath)
 		pe.Controller.FillBlockDeviceDetails(device)
 		// if ApplyFilter returns true then we process the event further
 		if !pe.Controller.ApplyFilter(device) {
 			continue
 		}
-		klog.Infof("Processed details for %s : %s", device.DevPath, device.UUID)
-		deviceInfo := pe.Controller.NewDeviceInfoFromBlockDevice(device)
+		klog.Infof("Processed details for %s", device.DevPath)
 
-		existingBlockDeviceResource := pe.Controller.GetExistingBlockDeviceResource(bdList, deviceInfo.UUID)
-		err := pe.Controller.PushBlockDeviceResource(existingBlockDeviceResource, deviceInfo)
-		if err != nil {
-			isErrorDuringUpdate = true
-			klog.Error(err)
+		// if GPTBasedUUID need to be used, generate the UUID,
```

Comment:
These comments probably should now be moved into the `addBlockDevice` method, as the below code doesn't show the generation.

Reply:
Done.